examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp

```diff
@@ -240,10 +251,14 @@ void OTAProviderExample::SendQueryImageResponse(app::CommandHandler * commandObj
         FabricInfo * fabricInfo = Server::GetInstance().GetFabricTable().FindFabricWithIndex(fabricIndex);
         NodeId nodeId           = fabricInfo->GetPeerId().GetNodeId();
 
-        // Only supporting BDX protocol for now
-        MutableCharSpan uri(uriBuf, kUriMaxLen);
-        chip::bdx::MakeURI(nodeId, CharSpan::fromCharString(mOTAFilePath), uri);
-        ChipLogDetail(SoftwareUpdate, "Generated URI: %.*s", static_cast<int>(uri.size()), uri.data());
+        // Generate the ImageURI if one is not already preset
+        if (strlen(mImageUri) == 0)
+        {
+            // Only supporting BDX protocol for now
+            MutableCharSpan uri(mImageUri, kUriMaxLen);
+            chip::bdx::MakeURI(nodeId, CharSpan::fromCharString(mOTAFilePath), uri);
```

Comment:
What happens if MakeURI returns error?  The string might not be null-terminated then...

Reply:
Good point. I opted to clear mImageUri in this case and send the response anyway and let the requestor deal with whether it should retry or just wait for the next query.